src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -120,6 +121,13 @@ private void grantPostponement(SscsCaseData sscsCaseData, PreSubmitCallbackRespo
         log.info("Action postponement request: postponement listingOption {} mapped to Event {} for case {}",
             listingOption, postponementEventType, sscsCaseData.getCcdCaseId());
 
+
+        if (listingOption.equalsIgnoreCase("readytolist")) {
+            sscsCaseData.setState(State.READY_TO_LIST);
+        } else if (listingOption.equalsIgnoreCase("notlistable")) {
+            sscsCaseData.setState(State.NOT_LISTABLE);
```

Comment:
you can simplify this with:
sscsCaseData.setState(State.valueOf(listingOption));
Doing this should also sort out your Code coverage (check sonarcube)